dbflute-runtime/src/main/java/org/dbflute/cbean/ckey/ConditionKey.java

```diff
@@ -449,10 +450,19 @@ protected ColumnRealName resolveCompoundColumn(ColumnRealName baseRealName, Cond
         }
         final List<SpecifiedColumn> compoundColumnList = option.getCompoundColumnList();
         final List<ColumnRealName> realNameList = new ArrayList<ColumnRealName>();
-        realNameList.add(baseRealName); // already cipher
-        for (SpecifiedColumn specifiedColumn : compoundColumnList) {
-            realNameList.add(doResolveCompoundColumn(option, specifiedColumn));
+        if (option.treatsCompoundColumnNullAsEmpty()) {
+            realNameList.add(toColumnRealName("coalesce(" + baseRealName + ",\'\')"));
+            for (SpecifiedColumn specifiedColumn : compoundColumnList) {
+                realNameList.add(toColumnRealName("coalesce(" + doResolveCompoundColumn(option, specifiedColumn) + ",\'\')"));
+            }
+        } else {
+            realNameList.add(baseRealName); // already cipher
+            for (SpecifiedColumn specifiedColumn : compoundColumnList) {
+                ColumnRealName resolvedColumnRealName = doResolveCompoundColumn(option, specifiedColumn);
+                realNameList.add(resolvedColumnRealName);      
+            }  
```

Comment:
I want you to refactor this logic if you can.
Please unify for-statement.

Reply:
> I want you to refactor this logic if you can.
> Please unify for-statement.

modified for-statement.

https://github.com/dbflute/dbflute-core/pull/54/commits/3178d878b1e2f463866a5dabcca86c2172091b4e